src/Microsoft.VisualStudio.Services.Agent/ConfigurationStore.cs

```diff
@@ -46,6 +46,12 @@ public sealed class AgentSettings
 
         [DataMember(EmitDefaultValue = false)]
         public int DeploymentGroupId { get; set; }
+
+        [DataMember(EmitDefaultValue = false)]
+        public string ProjectId { get; set; }
```

Comment:
1) support will not be forever. We will get the uses data and then need to decide if to remove the projectName from settings class
2) If we remove projectName support, existing customer will not be able to un-configure the agents as their agent setting file will still contain the projectName